File: Python/primenumber.py

def primeNumber(n):
    if (n==2):
        return True
    if n==1  or n%2==0:
        return False
    i=2
    while i*i<=n:
        if n%i==0:
            return False
        i+=1
    return True
if __name__=="__main__":
    n=int(input(""))
    if primeNumber(n):
        print(f"{n} is a prime number")
    else:
        print(f"{n} is not a prime number")

File: C++/primeNumber.cpp
#include<bits/stdc++.h>
using namespace std;


int main(void)
{
	int n,i,count=0;
	cin>>n;
	for(i=2;i<=(n/2);i++){
		if(n%i==0)
			count+=1;
	}
	if(count>0)
		cout<<"Not a prime number";
	else
		cout<<"A prime number";

}

